presto-main/src/main/java/com/facebook/presto/operator/ExchangeClient.java

```diff
@@ -506,4 +513,27 @@ private static void closeQuietly(HttpPageBufferClient client)
             // ignored
         }
     }
+
+    private static class ExponentialMovingAverage
+    {
+        private final double alpha;
+        private double oldValue;
+
+        public ExponentialMovingAverage(double alpha, double initialValue)
+        {
+            this.alpha = alpha;
+            this.oldValue = initialValue;
+        }
+
+        public void addSample(double value)
+        {
+            double newValue = oldValue + (alpha * (value - oldValue));
+            oldValue = newValue;
+        }
+
+        public long value()
```

Comment:
nit: `get()`? So the invocation looks like `averageBytesPerRequest.get()`?

Reply:
Changed to get